tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,153 @@
+import os
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.bluechi_is_online import BluechiIsOnline
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+DEFAULT_TIMEOUT = int(os.getenv("BC_IS_ON_WAIT_TIMEOUT", 1000))
+SHORT_TIMEOUT = int(os.getenv("BC_IS_ON_WAIT_TIMEOUT_SHORT", 5000))
+LONG_TIMEOUT = int(os.getenv("BC_IS_ON_WAIT_TIMEOUT_LONG", 20000))
+SLEEP_DURATION = int(os.getenv("SLEEP_DURATION", 2))
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_agent(
+    bluechi_is_online: BluechiIsOnline,
+    future: ResultFuture,
+    description: str,
+    wait_time: int = 0,
+    expected_result: int = 0,
+):
+    result = bluechi_is_online.agent_is_online(
+        wait_time,
+    )
+    future.result = result
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+
+    # Test 1: Start agent and verify --wait option returns 0 immediately
+    node_foo.systemctl.start_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "active")
+    LOGGER.debug("Starting test number 1 - agent should be online.")
+
+    result_future_online = ResultFuture()
+    start_time_immediately = time.time()
+    check_thread_online = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_online,
+            "Checking if agent is online",
+            DEFAULT_TIMEOUT,
+            0,
+        ),
+    )
+    check_thread_online.start()
+    check_thread_online.join()
+    elapsed_time = time.time() - start_time_immediately
+    assert (
+        elapsed_time < 1 and result_future_online.result
+    ), "Expected less than 1 second and the agent to be online"
+
+    # Test 2: Stop agent and verify --wait returns 1 after wait time is expired.
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Starting test number 2 - agent should remain offline.")
+
+    result_future_offline = ResultFuture()
+    start_time_wait_5_sec = time.time()
+    check_thread_offline = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_offline,
+            "Checking if agent remains offline",
+            SHORT_TIMEOUT,
+            1,
+        ),
+    )
+    check_thread_offline.start()
+    check_thread_offline.join()
+    elapsed_time_5_sec = time.time() - start_time_wait_5_sec
+    assert 5 < elapsed_time_5_sec < 6, "--wait returned offline after 5 seconds"
+    assert not result_future_offline.result, "Expected agent to remain offline"
```

Comment:
Actually, this is the same case as @mkemel mentioned above - we don't don anything in parallel and simply wait the `wait_time` interval for `bluechi-is-online` to exit. 
```python
wait_time = 5
result = False
start_time = time.time()
# extend the test timeout here a bit since we are wrapping bluechi-is-online
with Timeout(wait_time + 1, "bluechi-is-online did not exit within the specified time"):
    result = node_foo.bluechi_is_online(wait_time*1000) 
assert not result, f"Expected bluechi-is-online with --wait={wait_time} on node {NODE_FOO} to exit false" 
assert start_time - time.time() <= 5, f"Expected around 5 second for bluechi-is-online to exit"
 
```